src/tests/ftest/util/ior_test_base.py

```diff
@@ -285,97 +288,115 @@ def stop_ior(self):
-        # library
-        # Job2 will have 1 client set up to use only dfuse.
-        job1 = self.get_new_job(self.hostlist_clients[:-1], 1,
-                                results, intercept)
-        job2 = self.get_new_job([self.hostlist_clients[-1]], 2,
-                                results, None)
-
-        job1.start()
-        # Since same ior_cmd is used to trigger the MPIRUN
-        # with different parameters, pausing for 2 seconds to
-        # avoid data collisions.
-        time.sleep(2)
-        job2.start()
-        job1.join()
-        job2.join()
+        # Create two threads and run in parallel.
+        thread1 = self.create_ior_thread(
+            ior_cmd1, with_clients, 1, results, intercept)
+        thread2 = self.create_ior_thread(
+            ior_cmd2, without_clients, 2, results, None)
+
+        thread1.start()
+        thread2.start()
+        thread1.join()
+        thread2.join()
+
         self.stop_dfuse()
 
-    def get_new_job(self, clients, job_num, results, intercept=None):
+    def create_ior_thread(self, ior_command, clients, job_num, results,
+                          intercept=None):
         """Create a new thread for ior run.
 
         Args:
+            ior_command (IorCommand): IOR command instance.
             clients (list): hosts on which to run ior
             job_num (int): Assigned job number
             results (dict): A dictionary object to store the ior metrics
             intercept (path): Path to interception library
         """
-        job = threading.Thread(target=self.run_multiple_ior, args=[
-            clients, results, job_num, intercept])
+        job = threading.Thread(
+            target=self.run_custom_ior_cmd,
+            args=[ior_command, clients, results, job_num, intercept])
         return job
 
-    def run_multiple_ior(self, clients, results, job_num, intercept=None):
-        """Run the IOR command.
+    def run_custom_ior_cmd(self, ior_command, clients, results, job_num,
+                           intercept=None):
+        """Run customized IOR command, not self.ior_cmd.
+
+        Expected to be used with a threaded code where multiple IOR commands are
+        executed in parallel.
+
+        Display pool space before running it for a reference.
 
         Args:
+            ior_command (IorCommand): Custom IOR command instance.
             clients (list): hosts on which to run ior
             results (dict): A dictionary object to store the ior metrics
             job_num (int): Assigned job number
             intercept (str, optional): path to interception library. Defaults to
                 None.
         """
-        self.lock.acquire(True)
-        tsize = self.ior_cmd.transfer_size.value
-        testfile = os.path.join(self.dfuse.mount_dir.value,
-                                "testfile{}{}".format(tsize, job_num))
+        self.log.info("--- IOR Thread {}: Start ---".format(job_num))
```

Comment:
(pylint-logging-format-interpolation) Use lazy % formatting in logging functions